sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -38,24 +38,28 @@ class UnivocityParser(
     requiredSchema: StructType,
     val options: CSVOptions) extends Logging {
   require(requiredSchema.toSet.subsetOf(dataSchema.toSet),
-    "requiredSchema should be the subset of schema.")
+    "requiredSchema should be the subset of dataSchema.")
 
   def this(schema: StructType, options: CSVOptions) = this(schema, schema, options)
 
   // A `ValueConverter` is responsible for converting the given value to a desired type.
   private type ValueConverter = String => Any
 
+  // This index is used to reorder parsed tokens
+  private val tokenIndexArr =
```

Comment:
Could you convert it to `Array` explicitly. I have checked type of `tokenIndexArr`, it is actually `scala.collection.immutable.$colon$colon` - lazy list.

Reply:
Any side-effect?

Reply:
You have O(n) instead of O(1) for getting a value from the collection by an index.

Reply:
ah, I see. I'll recheck. Thanks!

Reply:
ok, fixed.